src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -0,0 +1,1931 @@
+
+        verifier.VerifyIL("<top-level-statements-entry-point>", """
+            {
+              // Code size       12 (0xc)
+              .maxstack  1
+              IL_0000:  call       "int[] Program.<<Main>$>g__arr|0_0()"
+              IL_0005:  newobj     "System.Span<int>..ctor(int[])"
+              IL_000a:  pop
+              IL_000b:  ret
+            }
+            """);
+    }
+
+    [Fact]
+    public void Conversion_Array_Span_Implicit_SemanticModel()
+    {
+        var source = """
+            class C
+            {
+                System.Span<int> M(int[] arg) { return arg; }
+            }
+            """;
+
+        var comp = CreateCompilationWithSpan(source);
+        var tree = comp.SyntaxTrees.Single();
+        var model = comp.GetSemanticModel(tree);
+
+        var arg = tree.GetRoot().DescendantNodes().OfType<ReturnStatementSyntax>().Single().Expression;
+        Assert.Equal("arg", arg!.ToString());
+
+        var argType = model.GetTypeInfo(arg);
+        Assert.Equal("System.Int32[]", argType.Type.ToTestDisplayString());
+        Assert.Equal("System.Span<System.Int32>", argType.ConvertedType.ToTestDisplayString());
+
+        var argConv = model.GetConversion(arg);
+        Assert.True(argConv.IsSpan);
+        Assert.True(argConv.IsImplicit);
+        Assert.False(argConv.IsUserDefined);
+        Assert.False(argConv.IsIdentity);
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void Conversion_Array_Span_Implicit_UnrelatedElementType(LanguageVersion langVersion)
+    {
+        var source = """
+            class C
+            {
+                System.Span<string> M(int[] arg) => arg;
+            }
+            """;
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion)).VerifyDiagnostics(
+            // (3,41): error CS0029: Cannot implicitly convert type 'int[]' to 'System.Span<string>'
+            //     System.Span<string> M(int[] arg) => arg;
+            Diagnostic(ErrorCode.ERR_NoImplicitConv, "arg").WithArguments("int[]", "System.Span<string>").WithLocation(3, 41));
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void Conversion_Array_Span_Implicit_NullableAnalysis(LanguageVersion langVersion)
+    {
+        var source = """
+            #nullable enable
+            class C
+            {
+                System.Span<string> M1(string[] arg) => arg;
+                System.Span<string> M2(string?[] arg) => arg;
+                System.Span<string?> M3(string[] arg) => arg;
+                System.Span<string?> M4(string?[] arg) => arg;
+                System.Span<int> M5(int?[] arg) => arg;
+                System.Span<int?> M6(int[] arg) => arg;
+                System.Span<int?> M7(int?[] arg) => arg;
```

Comment:
Are we testing nullable analysis in these cases? `int` and `int?` are distinct types.

Reply:
You're right, it's not nullable analysis, I added those because they seemed related, but we test "no conversion" elsewhere, so I will remove from here.